array/DNAm/analysis/EWAS/testCellComposition.r

```diff
@@ -0,0 +1,165 @@
+##---------------------------------------------------------------------#
+##
+## Title: Test cell type composition against disease status
+##
+## Purpose of script: calculate cellular composition of brain cell types
+## and test for differences between cases and controls
+##
+##---------------------------------------------------------------------#
+
+#----------------------------------------------------------------------#
+# DEFINE CETYGO FUNCTION FOR BRAIN TISSUE
+#----------------------------------------------------------------------#
+
+calcBrainComposition <- function(betas){
+  
+  predPropAll<-list() # store output in list
+
+  # run CETYGO using both ANOVA and IDOL methods and all available cell ref panels
+    for(method in names(modelBrainCoef)){
+        for(j in 1:length(modelBrainCoef[[method]])){
+        if(!is.null(modelBrainCoef[[method]][[j]])){
+                predPropAll[[method]][[j]]<-projectCellTypeWithError(betas, modelBrainCoef[[method]][[j]])
+        }
+        } 
+    }
+    predPropBest<-cbind(predPropAll[["ANOVA"]][[1]][,c("NeuNNeg_SOX10Neg", "NeuNPos")], predPropAll[["IDOL"]][[5]][,c("NeuNNeg_Sox10Neg_IRF8Pos","NeuNNeg_Sox10Neg_IRF8Neg")], predPropAll[["ANOVA"]][[3]][,c("SATB2Neg", "SATB2Pos")], predPropAll[["ANOVA"]][[6]][,c("NeuNNeg", "NeuNPos_SOX6Pos", "NeuNPos_SOX6Neg")], predPropAll[["IDOL"]][[4]][,"NeuNNeg_SOX10Pos"])
+    colnames(predPropBest)[10]<-"NeuNNeg_SOX10Pos"

```

Comment:
Why is this the only column name we are setting?

Reply:
Code later in the file suggests that the column names are being extracted during the `cbind` command above. Are they being extracted or not? Line 27 here isn't changing the name of the column either as it is the exact same name as the name in `predPropAll`. 
```R
datLong<-gather(predPropBest[,c("Cell.type", "Phenotype", "NeuNNeg_SOX10Neg","NeuNPos","NeuNNeg_Sox10Neg_IRF8Pos","NeuNNeg_Sox10Neg_IRF8Neg","SATB2Neg","SATB2Pos","NeuNNeg","NeuNPos_SOX6Pos","NeuNPos_SOX6Neg","NeuNNeg_SOX10Pos", "CETYGO")], Fraction, Proportion, NeuNNeg_SOX10Neg:CETYGO)
```

Reply:
It is a quirk with R. I think because we are just pulling across a single column at the end, it converts it to a 1D structure and loses the column heading so it needs to be re-added in.